### ISSUE TITLE
Add coercion rules for lists when used as input type

### ISSUE THREAD

**Issue by mehranj73** [apollographql/apollo-android#3336]
I get the following error for https://graphql.bitquery.io/:

> Can't parse `Array` value, expected array
> ----------------------------------------------------
> [2]:  ethereum(network: ethereum) {
> [3]:    transactions(txTo: {in:$address_list}) {
> ----------------------------------------------------
> 

GetEthBalance.graphql:
```
query GetEthBalance($address_list:[String!]!){
  ethereum(network: ethereum) {
    transactions(txTo: {in:$address_list}) {
      to{address}
      ntx:count
      received_balance: amount
    }
    address(address: {in:$address_list}) {
      address
      balance
    }
  }
}
```

**Comment by martinbonnin** [comment]
```graphql
transactions(txTo: {in:$address_list}) {
```

`txTo` is of type `[EthereumAddressSelector!]` so you'll need to provide an array. Same for `address`:

```graphql
query GetEthBalance($address_list:[String!]!){
  ethereum(network: ethereum) {
    transactions(txTo: [{in:$address_list}]) {
      to{address}
      ntx:count
      received_balance: amount
    }
    address(address: [{in:$address_list}]) {
      address
      balance
    }
  }
}
```

I'm not sure how come the explorer accepts this... I'll try to check if we're missing a coercion rule somewhere

**Comment by martinbonnin** [comment]
Indeed, there's a specific rule to coerce single values to lists containing a single element: https://spec.graphql.org/draft/#sec-List.Input-Coercion

We'll add that but that might take a bit of time. In the meantime, the above workaround should work.

**Comment by mehranj73** [comment]
It worked, thanks. Close the issue if you want. thanks again!

**Comment by martinbonnin** [comment]
Nice, thanks for the update 👍 . I'll keep the issue open as a reminder we need to implement these coercion rules.

**Comment by martinbonnin** [comment]
Closed in `3.0.0-alpha04`
